dotCMS/src/main/java/com/dotcms/rest/api/v1/taillog/TailLogResource.java

```diff
@@ -0,0 +1,214 @@
+import com.dotcms.rest.EmptyHttpResponse;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.servlets.taillog.Tailer;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.SecurityLogger;
+import com.dotmarketing.util.ThreadUtils;
+import com.dotmarketing.util.UtilMethods;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import org.glassfish.jersey.media.sse.EventOutput;
+import org.glassfish.jersey.media.sse.OutboundEvent;
+import org.glassfish.jersey.media.sse.SseFeature;
+import org.glassfish.jersey.server.JSONP;
+
+/**
+ * This resource provides the endpoint used by the LogViewer functionality to display backend server logs
+ * @author nollymarlonga
+ */
+@Path("/v1/tailLog")
+@Tag(name = "TailLog")
+public class TailLogResource {
+
+    @GET
+    @Path("/{fileName}")
+    @JSONP
+    @NoCache
+    @Produces(SseFeature.SERVER_SENT_EVENTS)
+    public final EventOutput getLogs(@Context final HttpServletRequest request,
+            @PathParam("fileName") final String fileName) throws IOException {
+
+        new WebResource().init(null, request, new EmptyHttpResponse(), true, null);
+
+        if(fileName.trim().isEmpty()) {
+            return sendError("File should not be empty");
+        }
+
+
+        String tailLogLofFolder = com.dotmarketing.util.Config.getStringProperty("TAIL_LOG_LOG_FOLDER", "./dotsecure/logs/");
+        if (!tailLogLofFolder.endsWith(java.io.File.separator)) {
+            tailLogLofFolder = tailLogLofFolder + java.io.File.separator;
+        }
+
+        final File logFolder 	= new File(com.dotmarketing.util.FileUtil.getAbsolutlePath(tailLogLofFolder));
+        final File logFile 	= new File(com.dotmarketing.util.FileUtil.getAbsolutlePath(tailLogLofFolder + fileName));
+
+
+        // if the logFile is outside of the logFolder, die
+        if ( !logFolder.exists()
+                ||   !logFile.getCanonicalPath().startsWith(logFolder.getCanonicalPath())) {
+
+            final String errorMessage = "Invalid File request:" + logFile.getCanonicalPath() + " from:" +request.getRemoteHost();
+            SecurityLogger.logInfo(
+                    TailLogResource.class,  errorMessage);
+            return sendError(errorMessage);
+        }
+
+        Logger.info(this.getClass(), "Requested logFile:" + logFile.getCanonicalPath());
+
+        final String regex = Config.getStringProperty("TAIL_LOG_FILE_REGEX", ".*\\.log$|.*\\.out$");
```

Comment:
depending on how often will be this endpoint called, it would be better to add a Lazy to avoid blocking issues

Reply:
This endpoint shouldn't be used by any functionality other than the TailLogResource. The TailLogResource is not meant to tail so many files at a time.